src/sentry/utils/cursors.py

```diff
@@ -201,8 +232,8 @@ def _build_prev_values(cursor, results, key, limit, is_desc):
     for result in result_iter:
         result_value = key(result, for_prev=True)
 
-        is_larger = result_value >= prev_value
-        is_smaller = result_value <= prev_value
+        is_larger = result_value >= prev_value  # type: ignore
+        is_smaller = result_value <= prev_value  # type: ignore
```

Comment:
Couldn't find a good way to resolve these issues. The problem is that the union doesn't compare well to itself since it contains types that are incompatible.

This part worked ok with `TypeVar("CursorValue", float, int, str)`, but that caused other issues